subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -761,8 +797,107 @@ static uint32_t adv_iso_start(struct ll_adv_iso_set *adv_iso,
 	return ret;
 }
 
+static uint8_t adv_iso_chm_update(uint8_t big_handle)
+{
+	struct ll_adv_iso_set *adv_iso;
+	struct lll_adv_iso *lll_iso;
+
+	adv_iso = adv_iso_get(big_handle);
+	if (!adv_iso) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	lll_iso = &adv_iso->lll;
+	if (lll_iso->term_req ||
+	    (lll_iso->chm_req != lll_iso->chm_ack)) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	/* Request channel map update procedure */
+	lll_iso->chm_chan_count = ull_chan_map_get(lll_iso->chm_chan_map);
+	lll_iso->chm_req++;
+
+	return BT_HCI_ERR_SUCCESS;
+}
+
+static void adv_iso_chm_complete_commit(struct lll_adv_iso *lll_iso)
+{
+	uint8_t hdr_data[ULL_ADV_HDR_DATA_LEN_SIZE +
+			 ULL_ADV_HDR_DATA_ACAD_PTR_SIZE];
+	struct pdu_adv *pdu_prev, *pdu;
+	struct lll_adv_sync *lll_sync;
+	struct pdu_big_info *bi;
+	struct ll_adv_set *adv;
+	uint8_t acad_len;
+	uint8_t ter_idx;
+	uint8_t ad_len;
+	uint8_t *acad;
+	uint8_t *ad;
+	uint8_t len;
+	uint8_t err;
+
+	/* Allocate next PDU */
+	adv = HDR_LLL2ULL(lll_iso->adv);
+	err = ull_adv_sync_pdu_alloc(adv, ULL_ADV_PDU_EXTRA_DATA_ALLOC_IF_EXIST,
+				     &pdu_prev, &pdu, NULL, NULL, &ter_idx);
+	LL_ASSERT(!err);
+
+	/* Get the size of current ACAD, first octet returns the old length and
+	 * followed by pointer to previous offset to ACAD in the PDU.
+	 */
+	lll_sync = adv->lll.sync;
+	hdr_data[ULL_ADV_HDR_DATA_LEN_OFFSET] = 0U;
+	err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu,
+					 ULL_ADV_PDU_HDR_FIELD_ACAD, 0U,
+					 &hdr_data);
+	LL_ASSERT(!err);
+
+	/* Dev assert if ACAD empty */
+	LL_ASSERT(hdr_data[ULL_ADV_HDR_DATA_LEN_OFFSET]);
+
+	/* Get the pointer, prev content and size of current ACAD */
+	err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu,
+					 ULL_ADV_PDU_HDR_FIELD_ACAD, 0U,
+					 &hdr_data);
+	LL_ASSERT(!err);
+
+	/* Find the BIGInfo */
+	acad_len = hdr_data[ULL_ADV_HDR_DATA_LEN_OFFSET];
+	len = acad_len;
+	(void)memcpy(&acad, &hdr_data[ULL_ADV_HDR_DATA_ACAD_PTR_OFFSET],
+		     sizeof(acad));
+	ad = acad;
+	do {
+		ad_len = ad[PDU_ADV_DATA_HEADER_LEN_OFFSET];
+		if (ad_len &&
+		    (ad[PDU_ADV_DATA_HEADER_TYPE_OFFSET] == BT_DATA_BIG_INFO)) {
+			break;
+		}
+
+		ad_len += 1U;
+
+		LL_ASSERT(ad_len <= len);
+
+		ad += ad_len;
+		len -= ad_len;
+	} while (len);
+	LL_ASSERT(len);
+
+	/* Get reference to BIGInfo */
+	bi = (void *)&ad[PDU_ADV_DATA_HEADER_DATA_OFFSET];
+
+	/* Copy the new/current Channel Map */
+	(void)memcpy(bi->chm_phy, lll_iso->data_chan_map, sizeof(bi->chm_phy));
+	bi->chm_phy[4] &= 0x1F;
+	bi->chm_phy[4] |= ((find_lsb_set(lll_iso->phy) - 1U) << 5);
```

Comment:
This clears some bits with &= and then sets them with |= right? 

Given that we do this other places as well, perhaps this could be packed into a macro or function?